### DISCUSSION TITLE
How does the Git integration option work exactly?

### DISCUSSION THREAD

**Discussion by geoidesic** [zotonic/zotonic#3953] [Q&A]
(Note: relates to https://github.com/zotonic/zotonic/discussions/3960)

I'm confused by how to integrate with git. This form is offered during site creation.

<img width="832" alt="Screenshot 2024-11-24 at 17 06 31" src="https://github.com/user-attachments/assets/839aa842-511a-439d-9aee-2bac22e49686">

From the .gitignore files it seems that its intention is to allow each client site – i.e. the `apps_user/<my_site>` – to be controlled via Git. I.e. these folders:

<img width="344" alt="Screenshot 2024-11-26 at 12 45 10" src="https://github.com/user-attachments/assets/cb0851e4-b095-4642-b45b-d15bbfdff1d2">

This sounds good. However, what is confusing is that, at this stage (i.e. creating the site), the site doesn't exist, so `checking out the site` would check out a blank site:

## Question 1
Is that expected? (i.e. that it would check out a blank, initialized repo)?

## Question 2
What does it mean where it says, "to add config files to the Git repo"? What config files?

## Question 3
This form only seems to be available during site creation. What if I want to attach it later?

## Question 4
Will this work with GitHub?

**Comment by mworrell** [comment]
The idea of the git integration was that you have a complete site at the Git URL and that that complete site is then checked out.

It might need some extra attention to make it more useful.

We assume that most people will work from the command line to commit their changes to their git repo.  That is why you don't have a similar form for an existing site.

**Reply by geoidesic** [reply]
I would suggest that you alter this section of the form in the following way. 

1. Since it is optional, make it start collapsed, in the same way that the Database section is collapsed. This intuitively tells the developer they don't need to worry about this section
2. Change the text to explain it better: firstly – "If you are creating a new site, ignore this section. If you have an existing Zotonic site on a git repo, you can optionally specify it here, in which case Zotonic will check that out instead of creating a new site. (Note that Zotonic only handles initial check out, after which it is expected that you will handle further git interactions yourself)"
3. I also suggest improving the message about config. I didn't understand that at all. It's very confusing... 
3a. It says "An existing config file will be overwritten" – why would there be an existing config file if you are creating a new site? 
3b. "To add config files to your git repo, add them to config.d directory of your site" – What sort of config files do you mean? 

Perhaps 3 is intended to mean this: 

"If you are using a git repository, then the zotonic site `*.config.*` files which are normally stored in `apps_user/<site_name>/priv/` should be moved to `apps_user/<site_name>/config.d`"? 

If so, say that :) If not, then please try to explain it to me and I'll help you phrase it better.

**Reply by mworrell** [reply]
I love your suggestions, am copying them over to the template.

**Reply by mworrell** [reply]
Added your suggestions here https://github.com/zotonic/zotonic/pull/3971
